### PR TITLE
List expression lite

### PR DESCRIPTION
**Pull request by rocky**
First cut at ListExpression
    
Many things don't work though. See comment-out code for avenues to explore.

@mmatera @TiagoCavalcante Feel free to hack away at this. 

<a href="https://gitpod.io/#https://github.com/Mathics3/mathics-core/pull/308"></a>

### CODE REVIEW

File: mathics/builtin/lists.py [hunk 1]
```diff
@@ -617,7 +618,7 @@ def list_boxes(items, f, open=None, close=None):
         sep = ","
     result = riffle(result, String(sep))
     if len(items) > 1:
-        result = Expression("RowBox", Expression(SymbolList, *result))
+        result = Expression(SymbolRowBox, Expression(SymbolList, *result))
```
Comment (mmatera) [inline mathics/builtin/lists.py:621]: With the current changes, this could be directly ``RowBox(*results)`` which avoids an evaluation step.

Comment (rocky) [reply mathics/builtin/lists.py:621]: Good! - Trying it now...

Comment (rocky) [reply mathics/builtin/lists.py:621]: No I tried it and I get a long ugly traceback when running: 

```
python -x mathics/docpipeline.py -s 'Formatting Output'
```

If you can get this to work, I'd be grateful.

Comment (mmatera) [reply mathics/builtin/lists.py:621]: #313 merge this with #311, which fixes this.

**Comment by rocky** [comment]
@mmatera given https://github.com/Mathics3/mathics-core/pull/313 which combines this and https://github.com/Mathics3/mathics-core/pull/311 (which ideally would have more changes). I would like to see this merged it and then https://github.com/Mathics3/mathics-core/pull/313 rebased off of this so that it just has the trailing changes and then that merged in.

In general, more smaller PRs are better than ones that combines two.

Your thoughts?

**Pull request merged**